src/components/InteractiveMap/InteractiveMap.vue

```diff
@@ -105,105 +52,125 @@
         .then(response => {
-          // Note that this will freeze the entire application, including
-          // progress spinners.
-          this.escherBuilder.load_map(response.data.map);
-          this.isLoadingMap = false;
+          this.mapData = response.data.map;
+        })
+        .catch(error => {
+          // TODO: show snackbar
         });
     },
-    reactionState(id: string, type?: string) {
-      // TODO
-    },
-    handleKnockout(reactionId: string) {
-      // TODO
-    },
-    handleKnockoutGenes(reactionId: string) {
-      // TODO
+    addDefaultCard() {
+      const model = this.$store.state.models.models.find(
+        model => model.name === "e_coli_core"
+      );
+      if (model) {
+        const organism = this.$store.getters["organisms/getOrganismById"](
+          model.organism_id
+        );
+        this.addCard("Design", organism, model, "pfba");
+      } else {
+        this.addCard("Design", null, null, "pfba");
+      }
     },
-    handleSetAsObjective(reactionId: string) {
-      // TODO
+    addCard(name, organism, model, method) {
+      const card = {
+        name: name,
+        organism: organism,
+        model: model,
+        method: method,
+        isSimulating: false,
+        growthRate: null,
+        isSelected: false
+      };
+      this.cards.push(card);
+      this.selectCard(card);
+      this.simulate(card);
     },
-    handleChangeBounds(reactionId: string, lower: string, upper: string) {
-      // TODO
+    removeCard(card) {
+      this.cards.splice(this.cards.indexOf(card), 1);
     },
-    handleResetBounds(reactionId: string) {
-      // TODO
+    selectCard(card) {
+      this.cards.forEach(card => {
+        card.isSelected = false;
+      });
+      card.isSelected = true;
     },
-    handleObjectiveDirection(reactionId: string) {
-      // TODO
+    simulate(card) {
+      if (card.model === null) {
+        // Cards are not guaranteed to have the model set (e.g. if the preferred
+        // default model doesn't exist - that could be the case for local
+        // installations of the platform).
+        return;
+      }
+
+      card.isSimulating = true;
+      axios
+        .post(`${settings.apis.model}/simulate`, {
+          model_id: card.model.id,
+          method: card.method
+        })
+        .then(response => {
+          card.growthRate = response.data.growth_rate;
+          // TODO: Flux distribution should belong to the card
+          this.fluxDistribution = response.data.flux_distribution;
+        })
+        .catch(error => {
+          // TODO: show snackbar
+          card.growthRate = null;
+          // TODO: Flux distribution should belong to the card
+          this.fluxDistribution = null;
+          console.error(error);
+        })
+        .then(response => {
+          card.isSimulating = false;
+        });
     }
   },
   computed: {
     maps() {
       return this.$store.state.maps.maps;
+    },
+    selectedCard() {
```

Comment:
With my proposed change above this would simply be coming from local state (data) `this.selectedCard`.